include/boost/geometry/algorithms/detail/disjoint/linear_areal.hpp

```diff
@@ -98,6 +115,66 @@ struct disjoint_no_intersections_policy<Geometry1, Geometry2, Tag1, multi_tag>
         }
         return true;
     }
+
+    typedef typename point_type<Geometry1>::type point1_type;
+
+    template <typename Strategy>
+    static inline bool apply(Geometry1 const& g1,
+                             Geometry2 const& g2,
+                             Strategy const& strategy,
+                             point1_type& p)
+    {
+        // TODO: use partition or rtree on g2
+        typedef typename boost::range_iterator<Geometry1 const>::type iterator;
+        for ( iterator it = boost::begin(g1) ; it != boost::end(g1) ; ++it )
+        {
+            typedef typename boost::range_value<Geometry1 const>::type value_type;
+            if ( ! disjoint_no_intersections_policy<value_type const, Geometry2>
+                    ::apply(*it, g2, strategy, p) )
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+};
+
+template<typename Geometry1, typename Geometry2,
+         typename NoIntersectionsPolicy
+                    = disjoint_no_intersections_policy
+                      <Geometry1, Geometry2> >
+struct disjoint_linear_areal_with_info
+{
+    typedef typename point_type<Geometry1>::type p_type;
+
+    typedef segment_intersection_points<p_type> intersection_return_type;
+
+    template <typename Strategy>
+    static inline intersection_return_type
```

Comment:
Right, so the problem is that the function `closest_points` takes the segment type which would define the result, but here and also in strategies the return type of a strategy is "derived" from the input geometries. This is probably not a problem in general since the coordinates are stored using these types anyway. So I think it's enough to calculate the coordinates and then convert to segment coordinates as you did. But the coordinate type of `p_type` could be the most precise coordinate type of both Geometry1 and Geometry2, not only Geometry1.

Reply:
ok, I am getting the most precise